src/main/java/vimification/taskui/CommandInput.java

```diff
@@ -79,17 +88,22 @@ private void executeCommand(String input) {
 
         try {
             CommandResult result = logic.execute(commandString);
-            parent.initializeTaskListPanel();
+            mainScreen.initializeTaskListPanel();
+            mainScreen.loadCommandResultComponent(result);
 
             // TODO: Should only clear if the task has been deleted.
             if (result.getFeedbackToUser().contains("Deleted Task:")) {
-                parent.clearRightComponent();
+                mainScreen.clearRightComponent();
             }
             System.out.println(result.getFeedbackToUser());
         } catch (CommandException e) {
             System.out.println("[Your command] " + input + " is invalid");
         } catch (ParserException e) {
-            System.out.println("[Your command] " + input + " can't be parsed");
+            // TODO: Remove this once execute no longer throws error and returns error in
+            // CommandResult
+            // instead.
+            CommandResult errorResult = new CommandResult("[Not a valid command] " + input);
```

Comment:
@VietAnh1010 `logic.execute()` still throws an error. Perhaps it is better if we wrap the error inside CommandResult and I can display it immediately?

Reply:
Agree, the `CommandResult` will be similar to an `Either` - it contains a string in case of error.